controllers/main.go

```diff
@@ -79,10 +80,13 @@ func randToken() string {
 // empty, http.Request.RemoteAddr. See the sample nginx.conf for using the
 // real_ip module to correctly set the X-Real-IP header.
 func getClientIP(r *http.Request) string {
-	getHost := func(ip string) string {
-		if strings.Contains(ip, ":") {
-			parts := strings.Split(ip, ":")
-			return parts[0]
+	// getHost returns the host portion of a string containing either a
+	// host:port formatted name or just a host. An empty string is returned if
+	// net.SplitHostPort returns an error.
+	getHost := func(hostPort string) string {
+		ip, _, err := net.SplitHostPort(hostPort)
+		if err != nil {
+			return ""
```

Comment:
In this case there are two possible sources: (1) whatever is in the `X-Real-IP` header or (2) the `RemoteAddr` field of the request as a backup.

The first is set by the proxy/load-balancer/whatever and 2 comes from golang's net library and is presumably set to a valid value, probably whatever interface it's bound to or server the request from.